### PR TITLE
chore: refactor Framer transport

### PR DESCRIPTION
**Pull request by nemith**
- Some cleanup of the transport code to be more straight forward and clear.  
- Fixed the testing and benchmarks so that they are benchmarking and checking for more corner cases.  
- Fixes some potential overflow bugs on 32bit platforms (someone trying to do something gross with this library on a 32-but arm box?  Maybe?) 
- 

```
❯ go test -bench=. -benchmem
goos: darwin
goarch: arm64
pkg: nemith.io/netconf/transport
cpu: Apple M4 Max
BenchmarkChunkedRead/Small_200B-16              25781038                43.57 ns/op     4819.58 MB/s           0 B/op          0 allocs/op
BenchmarkChunkedRead/Medium_128KB-16              324566              3693 ns/op        35557.35 MB/s          0 B/op          0 allocs/op
BenchmarkChunkedRead/Large_10MB-16                  3777            318602 ns/op        32968.05 MB/s          2 B/op          0 allocs/op
BenchmarkChunkedReadByte/Small_200B-16           2746282               450.7 ns/op       465.90 MB/s           0 B/op          0 allocs/op
BenchmarkChunkedReadByte/Medium_128KB-16            4563            272647 ns/op         481.58 MB/s           0 B/op          0 allocs/op
BenchmarkMarkedRead/Small_200B-16               70238932                17.38 ns/op     11853.29 MB/s          0 B/op          0 allocs/op
BenchmarkMarkedRead/Medium_128KB-16             66760936                17.47 ns/op     7502101.15 MB/s        0 B/op          0 allocs/op
BenchmarkMarkedRead/Large_10MB-16               59576182                17.47 ns/op     600381115.45 MB/s              0 B/op          0 allocs/op
BenchmarkMarkedReadByte/Small_200B-16           221047735                5.506 ns/op    37411.88 MB/s          0 B/op          0 allocs/op
BenchmarkMarkedReadByte/Medium_128KB-16         216967293                5.616 ns/op    23340891.18 MB/s               0 B/op          0 allocs/op
PASS
ok      nemith.io/netconf/transport     14.369s
```

### CODE REVIEW

File: transport/frame.go [hunk 1]
```diff
@@ -261,73 +270,108 @@ func (r *chunkReader) ReadByte() (byte, error) {
...
+
+		// Write payload
+		n, err := w.w.Write(p[:chunkSize])
+		totalWritten += n
+		if err != nil {
+			return totalWritten, err
+		}
+
+		// Advance
+		p = p[n:]
```
[the remark this answers is not in the document]
Comment (nemith) [reply transport/frame.go]: bufio.Writer doesn't do partial writes so this isn't necessary.
